src/test/java/kitchenpos/application/ProductServiceTest.java

```diff
@@ -92,38 +90,57 @@ void changePriceTest() {
         // given
         Product product = TEST_PRODUCT();
         Product changeProduct = TEST_PRODUCT();
-        changeProduct.setPrice(new BigDecimal(900));
+        changeProduct.setPrice(new BigDecimal(1500));
         UUID productId = product.getId();
         given(productRepository.findById(productId)).willReturn(Optional.of(product));
-        given(menuRepository.findAllByProductId(productId)).willReturn(List.of(TEST_MENU()));
 
         // when
+        Menu menu = TEST_MENU();
+        MenuProduct menuProduct = TEST_MENU_PRODUCT();
+        menuProduct.setProduct(changeProduct);
+        menuProduct.setProductId(changeProduct.getId());
+        menuProduct.setQuantity(2);
+        menu.setMenuProducts(List.of(menuProduct));
+        given(menuRepository.findAllByProductId(productId)).willReturn(List.of(menu));
+
         Product result = productService.changePrice(productId, changeProduct);
 
         // then
         verify(productRepository, times(1)).findById(productId);
         verify(menuRepository, times(1)).findAllByProductId(productId);
+        List<Menu> allByProductId = menuRepository.findAllByProductId(productId);
+
         assertThat(result.getPrice()).isEqualTo(changeProduct.getPrice());
+        assertThat(allByProductId).extracting("displayed").containsExactly(true);
     }
 
-    // TODO: 아직 작동안할 듯?
     @Test
     @DisplayName("가격 변경 후, 메뉴의 가격이 (메뉴에 포함된 상품들의 가격 x 개수) 총 합보다 높다면 메뉴를 비활성화 한다.")
     void changePriceAndHideTest() {
         // given
         Product product = TEST_PRODUCT();
         Product changeProduct = TEST_PRODUCT();
-        changeProduct.setPrice(new BigDecimal(100));
+        changeProduct.setPrice(new BigDecimal(10));
         UUID productId = product.getId();
         given(productRepository.findById(productId)).willReturn(Optional.of(product));
-        given(menuRepository.findAllByProductId(productId)).willReturn(List.of(TEST_MENU()));
 
         // when
+        Menu menu = TEST_MENU();
+        MenuProduct menuProduct = TEST_MENU_PRODUCT();
+        menuProduct.setProduct(changeProduct);
+        menuProduct.setProductId(changeProduct.getId());
+        menuProduct.setQuantity(2);
+        menu.setMenuProducts(List.of(menuProduct));
+        given(menuRepository.findAllByProductId(productId)).willReturn(List.of(menu));
+
         Product result = productService.changePrice(productId, changeProduct);
 
         // then
         verify(productRepository, times(1)).findById(productId);
         verify(menuRepository, times(1)).findAllByProductId(productId);
+        List<Menu> allByProductId = menuRepository.findAllByProductId(productId);
+
         assertThat(result.getPrice()).isEqualTo(changeProduct.getPrice());
+        assertThat(allByProductId).extracting("displayed").containsExactly(false);
```

Comment:
아래 처럼도 바꿀수 있을 것 같네요.
allByProductId 를 사용한 이유가 있을까요?
여기에 객체 생성하는 부분들을 Fixture 로 빼면 더 좋을 것 같네요.

```suggestion
        // given
        Product originalProduct = TEST_PRODUCT();

        Product changeProduct = TEST_PRODUCT();
        changeProduct.setPrice(new BigDecimal(10));

        MenuProduct menuProduct = TEST_MENU_PRODUCT();
        menuProduct.setProduct(changeProduct);
        menuProduct.setProductId(changeProduct.getId());
        menuProduct.setQuantity(2);

        Menu menu = TEST_MENU();
        menu.setMenuProducts(List.of(menuProduct));

        given(productRepository.findById(any())).willReturn(Optional.of(originalProduct));
        given(menuRepository.findAllByProductId(any())).willReturn(List.of(menu));

        assertThat(menu.isDisplayed()).isTrue();

        // when
        Product result = productService.changePrice(originalProduct.getId(), changeProduct);

        // then
        assertThat(result.getPrice()).isEqualTo(changeProduct.getPrice());
        assertThat(menu.isDisplayed()).isFalse();
```

Reply:
given으로  넘겨준 값이 서비스에서 changePrice를 통해 변경되지 않는다고 생각해서 다시 가져오는 테스트를 짰네요.

막상 또  given으로 다시 가져와서 쓰는 것 보면... 괴상한 코드네요.
``` given(menuRepository.findAllByProductId(productId)).willReturn(List.of(menu)) ```